core-processor/src/configs.rs

```diff
@@ -125,3 +131,53 @@ impl ExecutionSettings {
         self.allocations_config.load_page_cost
     }
 }
+
+/// Stable parameters for the whole block across processing runs
+#[derive(Clone)]
+pub struct BlockConfig {
+    /// Block info
+    pub block_info: BlockInfo,
+    /// Allocations config
+    pub allocations_config: AllocationsConfig,
+    /// Existential deposit
+    pub existential_deposit: u128,
+    /// Gas allowance
+    pub gas_allowance: u64,
+    /// Outgoing limit
+    pub outgoing_limit: u32,
+    /// Host function weights
+    pub host_fn_weights: HostFnWeights,
+    /// Forbidden functions
+    pub forbidden_funcs: BTreeSet<&'static str>,
+    /// Mailbox threshold
+    pub mailbox_threshold: u64,
+}
+
+impl Default for BlockConfig {
+    fn default() -> Self {
+        Self {
+            block_info: Default::default(),
+            allocations_config: Default::default(),
+            existential_deposit: EXISTENTIAL_DEPOSIT,
+            gas_allowance: u64::MAX,
+            outgoing_limit: OUTGOING_LIMIT,
+            host_fn_weights: Default::default(),
+            forbidden_funcs: Default::default(),
+            mailbox_threshold: MAILBOX_THRESHOLD,
+        }
+    }
+}
+
+/// Unstable parameters for message execution across processing runs
+#[derive(Clone)]
+pub struct MessageExecutionConfig {
+    /// Optional executable actor
+    pub executable_actor: Option<ExecutableActor>,
+    /// Incoming dispatch
+    pub dispatch: IncomingDispatch,
+    /// The external origin for a key
+    pub origin: ProgramId,
+    // TODO: Temporary here for non-executable case. Should be inside executable actor, renamed to Actor.
+    /// Destination program
+    pub program_id: ProgramId,
```

Comment:
resolve this todo, please, append id into ExecutableActor
in general, we should pass here just `Actor`, checking if it executable afterward